internal/pkg/cli/local_run.go

```diff
@@ -148,21 +145,35 @@ func newLocalRunOpts(vars localRunVars) (*localRunOpts, error) {
 		o.repository = repository.NewWithURI(ecr.New(defaultSessEnvRegion), repoName, resources.RepositoryURLs[o.wkldName])
 		return nil
 	}
-	opts.buildContainerImages = func(o *localRunOpts) error {
+	opts.buildContainerImages = func(o *localRunOpts, mft manifest.DynamicWorkload) (map[string]string, error) {
 		gitShortCommit := imageTagFromGit(o.cmd)
 		image := clideploy.ContainerImageIdentifier{
 			GitShortCommitTag: gitShortCommit,
 		}
-		return clideploy.BuildContainerImages(&clideploy.ImageActionInput{
+		out := &clideploy.UploadArtifactsOutput{}
+		err := clideploy.BuildContainerImages(&clideploy.ImageActionInput{
 			Name:               o.wkldName,
 			WorkspacePath:      o.ws.Path(),
 			Image:              image,
-			Mft:                o.appliedDynamicMft.Manifest(),
+			Mft:                mft.Manifest(),
 			GitShortCommitTag:  gitShortCommit,
 			Builder:            o.repository,
 			Login:              o.repository.Login,
 			CheckDockerEngine:  o.dockerEngine.CheckDockerEngineRunning,
-			LabeledTermPrinter: o.labeledTermPrinter}, &o.out)
+			LabeledTermPrinter: o.labeledTermPrinter,
+		}, out)
+		if err != nil {
+			return nil, err
+		}
+
+		containerURIs := make(map[string]string, len(out.ImageDigests))
+		for name, info := range out.ImageDigests {
+			if len(info.RepoTags) == 0 {
+				return nil, fmt.Errorf("no repo tags for image %q", name)
```

Comment:
In what scenario do users get this error?

Reply:
realistically i'm not aware of any bc `docker build` will fail if it isn't given any `repo:tag` pairs - so it's really only if there's some bug in that function (context: https://github.com/aws/copilot-cli/pull/5182#discussion_r1293980625)

Reply:
Sounds good. Maybe add a comment 🤔